internal/clients/ec2/ec2_client.go

```diff
@@ -122,6 +122,25 @@ func (c *Client) ListInstanceTypes() ([]types.InstanceTypeInfo, error) {
 	return resp.InstanceTypes, nil
 }
 
+func (c *Client) ListInstanceTypesWithPaginator() ([]types.InstanceTypeInfo, error) {
+	log.Trace().Msg("Listing AWS EC2 instance types Using Paginator")
+
+	input := &ec2.DescribeInstanceTypesInput{MaxResults: aws.Int32(100)}
+	pag := ec2.NewDescribeInstanceTypesPaginator(c.ec2, input)
+
+	res := make([]types.InstanceTypeInfo, 0)
```

Comment:
I would probably set the initial capacity to 64 or 128, almost always the runtime will have to reslice this from 8 to 16 to 32 and to 64. It smells like preliminary optimalization but it looks like amount of instances will be more than 100 all the time (it is currently 230 or so).